### PR TITLE
storage: Add missing mtGetAttr handlers

### PR DESCRIPTION
**Pull request by agkaminski**
## Description

lseek expects atSize attribute be available on block devices via mtGetAttr message. Fix missing handlers

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (refactoring, style fixes, git/CI config, submodule management, no code logic changes)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: storage/gr716-flash/flashsrv.c [hunk 1]
```diff
@@ -387,6 +387,16 @@ static void flashsrv_rawThread(void *arg)
 				mutexUnlock(memory->lock);
 				break;
 
+			case mtGetAttr:
```
Comment (lukileczo) [inline storage/gr716-flash/flashsrv.c:390]: I'd add raw IO check (applies also for `imxrt-flash`)

```suggestion
			case mtGetAttr:
				if (flashsrv_verifyRawIO(memory, &msg, &msg.o.size) < 0) {
					msg.o.err = -EINVAL;
					break;
				}
```

File: storage/gr716-flash/flashsrv.c [hunk 2]
```diff
@@ -430,6 +440,16 @@ static void flashsrv_devThread(void *arg)
 				msg.o.err = EOK;
 				break;
 
+			case mtGetAttr:
+				if (msg.i.attr.type == atSize) {
+					msg.o.attr.val = memory->ctx.properties->totalSz;
+					msg.o.err = 0;
```
Comment (gerard5) [inline storage/gr716-flash/flashsrv.c:446]: nit: both implementations use `EOK` for success instead of `0`

File: storage/imxrt-flash/flashsrv.c [hunk 3]
```diff
@@ -534,6 +534,16 @@ static void flashsrv_rawThread(void *arg)
 				msg.o.err = flashsrv_bufferedSync(memory->fOid.id);
 				break;
 
+			case mtGetAttr:
+				if (msg.i.attr.type == atSize) {
+					msg.o.attr.val = memory->parts[msg.oid.id].pHeader->size;
```
Comment (gerard5) [inline storage/imxrt-flash/flashsrv.c:539]: Oh, I thing `msg.oid.id` should be checked before `switch` if it's valid (`memory->parts[]` potential out of bounds access) when a message with an invalid `oid.id` would be sent.

File: storage/gr716-flash/flashsrv.c (hunk 1 above)
Comment (agkaminski) [reply storage/gr716-flash/flashsrv.c]: I moved id check to the common code

File: storage/gr716-flash/flashsrv.c (hunk 2 above)
Comment (agkaminski) [reply storage/gr716-flash/flashsrv.c:446]: EOK is non-standard and it alas leaked from the kernel. I removed EOKs from this function instead

File: storage/imxrt-flash/flashsrv.c (hunk 3 above)
Comment (agkaminski) [reply storage/imxrt-flash/flashsrv.c:539]: Thanks, done

File: storage/imxrt-flash/flashsrv.c [hunk 4]
```diff
@@ -559,32 +569,46 @@ static void flashsrv_devThread(void *arg)
...
-				break;
-
-			case mtSync:
-				msg.o.err = flashsrv_bufferedSync(memory->fOid.id);
-				break;
-
-			default:
-				msg.o.err = -ENOSYS;
-				break;
+		if (msg.oid.id >= memory->pCnt) {
```
Comment (lukileczo) [inline storage/imxrt-flash/flashsrv.c:572]: I think you mixed up the threads (here we base on `memory->fOid.id`) and in `flashsrv_rawThread` the id is read from msg.

Comment (agkaminski) [reply storage/imxrt-flash/flashsrv.c:492]: Right, thank you

**Review by lukileczo — APPROVED** [review]
LGTM

**Pull request merged**